omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/SessionController1_8.java

```diff
@@ -57,12 +65,41 @@
 	public Object get(WebRequest request) {
 		boolean authenticated = Context.isAuthenticated();
 		SimpleObject session = new SimpleObject();
+		
+		UserContext userContext = Context.getUserContext();
+		Provider currentProvider = null;
+		if (userContext != null && userContext.getAuthenticatedUser() != null) {
+			User currentUser = userContext.getAuthenticatedUser();
+			Collection<Provider> providers = new HashSet<Provider>();
+			try {
+				Context.addProxyPrivilege(PrivilegeConstants.VIEW_PROVIDERS);
+				if (currentUser.getPerson() != null) {
+					providers = Context.getProviderService().getProvidersByPerson(currentUser.getPerson(), false);
+				}
+			}
+			finally {
+				Context.removeProxyPrivilege(PrivilegeConstants.VIEW_PROVIDERS);
+			}
+			if (providers.size() > 1) {
+				throw new IllegalStateException("Can't handle users with multiple provider accounts");
+			} else if (providers.size() == 1) {
+				currentProvider = providers.iterator().next();
+			}
```

Comment:
I feel this section should go to a separate method "getCurrentProvider". Also please write tests for your changes